### PR TITLE
feat: Add page-updated event

### PR DESCRIPTION
**Pull request by asafshen**
## Related Issues

https://github.com/descope/etc/issues/4136

## Description

Add ready event with so developer-user can listen and show loading / any other purpose

### CODE REVIEW

**Comment by asafshen** [comment]
loading example on fast 3G network:
https://github.com/descope/descope-js/assets/10514677/b6411316-1913-4462-a910-ea03deb23f3d

File: packages/web-component/README.md [hunk 1]
```diff
@@ -143,3 +143,17 @@ Usage example:
 const descopeWcEle = document.getElementsByTagName('descope-wc')[0];
 descopeWcEle.addEventListener('success', (e) => alert(`Success! - ${JSON.stringify(e.detail)}`));
 ```
+
+### `page-ready` - Fired when the page is ready.
```
[the remark this answers is not in the document]
Comment (nirgur) [reply packages/web-component/README.md:147]: I think we should go with `ready`

File: packages/web-component/src/app/index.html [hunk 2]
```diff
@@ -59,6 +63,15 @@
       descopeWcEle.addEventListener('success', (e) =>
         alert(`Success! - ${JSON.stringify(e.detail)}`),
       );
+      descopeWcEle.addEventListener('page-ready', () => {
+        // Remove loading
+        const loading = document.querySelector('.loading');
+        if (loading) {
+          loading.remove();
```
Comment (nirgur) [inline packages/web-component/src/app/index.html:70]:
```
document.querySelector('.loading')?.remove();
```

**Review by nirgur — COMMENTED** [review]
Looks good
WDYT about changing the name to `ready`?

**Comment by asafshen** [comment]
> Looks good WDYT about changing the name to `ready`?

I'm in favor 👍 changing

File: packages/web-component/src/app/index.html (hunk 2 above)
Comment (asafshen) [reply packages/web-component/src/app/index.html]: thanks 👍

**Review by nirgur — APPROVED** [review]

**Pull request merged**